docs/reference/glossary/network-latency-management.md

```diff
@@ -0,0 +1,112 @@
+---
+id: network-latency-management
+title: Network Latency Management
+sidebar_label: Network Latency Management
+---
+
+Every player is affected by a certain amount of latency, which is the result of the distance to the server/host, the number of hops a packet has to make, and the server's tick rate and update rate.
+
+This is where Network Latency Management comes in. Due to these delays, you would see where a player was a few milliseconds ago, not where they are right now. If the game didn't compensate for at least a portion of this delay, then you would be unable to hit other players.
+
+
+
+## Server rewind
+
+The server uses the latency of the player (including any inherent delay due to interpolation) to rewind time by an appropriate amount in order to determine what the shooting client saw at the time the shot was fired.
+ 
+
+## Interpolation
+
+A ------> B
+
+Interpolation works by essentially buffering a game state and rendering the game state to the player with a small, constant delay. When a packet from the server arrives, before immediately updating the position of an object , the client will interpolate the position, starting from the last known position. 
+
+Over an interpolation interval, the object will be rendered  smoothly  moving between the two positions. 
+
+## Extrapolation
+
+
+A ---- B -----> 
+
+Extrapolation is an attempt to estimate a future game state. On receipt of a packet from the server, the position of an object is updated to the new position. Awaiting the next update, the next position is extrapolated based on the current position and the movement at the time of the update. 
+
+The client will normally assume that a moving object will continue in the same direction. When a new packet is received, the position may be updated.
+
+## Action casting
+
+Server driven action, but with client-side “casting” animations and sounds that respond to client inputs. This is to get responsiveness while still keeping the action server driven.
+
+## Client prediction:
+
+- **Deterministic lockstep:**
+  A method of networking a system from one computer to another by sending only the inputs that control that system, rather than the state of that system
+
+- **Deterministic rollback:**
+  An enhancement of deterministic lockstep where clients forward-predict inputs while waiting for updates. This setup is more complex but enables a more responsive game than lockstep. It’s relatively inexpensive and secure, but with very complex determinism and simulation.
+
+- **Input prediction:**
+  (and reconciliation to correct differences between client and server) TODO
+
+- **Ghost prediction**
```

Comment:
Industry term would be "Client Side Prediction". Ghost prediction is an alternative term used by DOTS and very few people.

Reply:
Doc Updated